types/serverless/plugins/aws/provider/awsProvider.d.ts

```diff
@@ -81,11 +81,16 @@ declare namespace Aws {
     }
 
     interface IamSettings {
-        role?: string;
+        role?: string | IamLogicalRole;
+        deploymentRole?: string;
+    }
+    
+    interface IamLogicalRole {
```

Comment:
Any reason why you've called this `IamLogicalRole` instead of just `IamRole`?

Reply:
Not really. 

When I looked at the serverless.yml reference given by Serverless they have a comment that seems to call this a logical role. I can change this for `IamRole` as it does make more sense.

serverless.yml reference:
```yml
  iam:
    # Overwrite the default IAM role which is used for all functions
    role: arn:aws:iam::XXXXXX:role/role
    # .. OR configure logical role
    role:
```